### ISSUE TITLE
For Observe-Composite operation, Why does the client report multiple items repeatedly?

### ISSUE THREAD

**Issue by dellzhui** [eclipse/leshan#1337]
I tried to find instructions on how the client should notation in the following document：
> [OMA-TS-LightweightM2M_Transport-V1_1-20180612-C](http://www.openmobilealliance.org/release/lightweightm2m/V1_1-20180612-C/OMA-TS-LightweightM2M_Transport-V1_1-20180612-C.html)
> [OMA-TS-LightweightM2M_Core-V1_1-20180612-C](http://openmobilealliance.org/RELEASE/LightweightM2M/V1_1-20180612-C/OMA-TS-LightweightM2M_Core-V1_1-20180612-C.pdf)

However, no results were found.

**[here](https://jijiantuku-image.oss-cn-beijing.aliyuncs.com/18-2-26/20221021.pcapng) is the pacp file.**

The specific problems I encountered are as follows：

1. server did Observe-Composite operation, it observed /20221/0/2 and /20221/0/3 and /20221/0/6
> ![](https://jijiantuku-image.oss-cn-beijing.aliyuncs.com/markdown_img/2022/10/21/20221021_162352_2aaf3e9282b5b2a2af21dc0b638df3e1.png)

2. client response 3 times:
> ![](https://jijiantuku-image.oss-cn-beijing.aliyuncs.com/markdown_img/2022/10/21/20221021_161611_7752d98c2bb6b411747a06536ded5e28.png)

3. The client reports every three seconds, but each time three packets are reported consecutively and the content is the same(Each package contains the values /2, /3, /6 at the same time)
> ![](https://jijiantuku-image.oss-cn-beijing.aliyuncs.com/markdown_img/2022/10/21/20221021_162515_980c6194d79fa23df4336b31acfb27dd.png)

**Comment by dellzhui** [comment]
[here](https://jijiantuku-image.oss-cn-beijing.aliyuncs.com/18-2-26/20221021.pcapng
) is the pacp file

**Comment by sbernard31** [comment]
By default, leshan client demo used only object 0, 1 , 3, 6, 3303 and 3442. there is no 20221.
So it's hard to understand where is come from ?

Don't you think it could help if you give some simple information like : 
- Is leshan used at client or server side ?
- If you don't use leshan for one peer,  what implementation are you using ?
- When you are using Leshan, which version ? are you using demo or something you code with leshan library or demo modified ?

**Comment by sbernard31** [comment]
I was understanding that you tried to report a kind of bug in Leshan but I just see that you edit your comment by adding : 

> I tried to find instructions on how the client should notation in the following document：

So maybe you ask about how to understand the LWM2M specification ?  Or you face a client (not leshan) with strange behavior and you want to know if this is an expected behavior or not ? 

Sorry, I'm a bit lost about what you  are asking  for :sweat_smile:

**Comment by dellzhui** [comment]
Thanks for your reply.

> So maybe you ask about how to understand the LWM2M specification ? Or you face a client (not leshan) with strange behavior and you want to know if this is an expected behavior or not ?

I implemented a lwm2m client based on leshan, and I don't think it's a bug in leshan.

What I am curious about is that for the Observe-Composite, it is very strange from the packet capture, and the client has reported it 3 same packages. I want to know if this is the logic of the leshan framework or a bug in the BaseInstanceEnabler subclass I implemented.

Now I'm pretty sure it's a problem with my own BaseInstanceEnabler subclass, it may execute fireResourcesChange method multiple times causing this issue.

still confirming...
